magento/magento2
# Magento Framework message-queue change suggestion

The Magento_MessageQueue module implements functionality to RabbitMQ service as described on the https://devdocs.magento.com/guides/v2.3/install-gde/prereq/install-rabbitmq.html page. I've made some codebase updates that are going to allow proper error display when Magento_Amqp disabled and PHP error displayed which is happening now.

### Description (*)
Scenario. Keep Magento_Amqp and Magento_AmqpStore modules disabled, add/config env.php as per https://devdocs.magento.com/guides/v2.3/install-gde/prereq/install-rabbitmq.html instruction, and try to run consumer queue manually.

Output:
$ php bin/magento queue:consumer:start async.operations.all
In ErrorHandler.php line 61:
Warning: Invalid argument supplied for foreach() in /srv/releases/20191031062112/vendor/magento/framework-message-queue/ConnectionTypeResolver.php on line 35

Expected results:
$ php bin/magento queue:consumer:start async.operations.all
In ConnectionTypeResolver.php line 43:
                                
  Unknown connection name amqp 

### Fixed Issues (if relevant)
file lib/internal/Magento/Framework/MessageQueue/ConnectionTypeResolver.php or vendor/magento/framework-message-queue/ConnectionTypeResolver.php updated with the following:

if(is_array($this->resolvers)){
        foreach ($this->resolvers as $resolver) {
            $type = $resolver->getConnectionType($connectionName);
            if ($type != null) {
                break;
            }
}
        }

        if ($type === null) {
            throw new \LogicException('Unknown connection name ' . $connectionName . ' check if Magento_Amqp module enabled');
        }
        return $type;
    }
}



### Manual testing scenarios (*)
1. Magento 2.3.2 and 2.3.3 disable Magento_Amqp and Magento_AmqpStore
2. Configure env.php file with https://devdocs.magento.com/guides/v2.3/install-gde/prereq/install-rabbitmq.html instruction.
3. Fire up one consumer, example:
php bin/magento queue:consumer:start async.operations.all

Expected results:
In ConnectionTypeResolver.php line 43:
  Unknown connection name amqp

Real results:
In ErrorHandler.php line 61:
Warning: Invalid argument supplied for foreach() in vendor/magento/framework-message-queue/ConnectionTypeResolver.php on line 35

### Questions or comments
Replicated/fixed on Magento 2.3.2 and 2.3.3 EE and CE

### Contribution checklist (*)
